src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -773,6 +774,41 @@ private static void CheckValidScopedMethodConversion(SyntaxNode syntax, MethodSy
             }
         }
 
+        private static void CheckLambdaConversion(LambdaSymbol lambdaSymbol, TypeSymbol targetType, BindingDiagnosticBag diagnostics)
+        {
+            // Warn for defaults/`params` mismatch.
+            if (!lambdaSymbol.SyntaxNode.IsKind(SyntaxKind.AnonymousMethodExpression))
```

Comment:
nit: consider inverting this `if` to flatten the method (bail out early)

Reply:
Thanks, changed: 2ac5fe13a808a4c236287cf9a6434a7d48027631